### PR TITLE
VAM-741: Add/Tag "SESSION" RPC Models

### PR DESCRIPTION
**Pull request by creindle**
[X] Completed creating model stubs
[X] Model validity tests pass
[X] RPC Router displays both: type `SESSION` and dataSource `VISTA`

### CODE REVIEW

File: emulation/models/xus-av-code.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const RPC_EMULATOR_MODEL = [{
+    id: 'XUS_AV_CODE',
+    name: 'XUS AV CODE',
+    function: 'VALIDAV^XUSRB',
+    parameters: [],
+    labels: ['SESSION', 'VISTA-ONLY'],
+    domain: '',
+    files: [''],
+    services: [''],
+    rpcType: 'UTILITY',
```
Comment (creindle) [inline emulation/models/xus-av-code.js]: I didn't know what the rpcTypes were for these, is this correct? I set them all to `UTILITY`

File: emulation/models/xus-esso-validate.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const RPC_EMULATOR_MODEL = [{
+    id: 'XUS_ESSO_VALIDATE',
+    name: 'XUS ESSO VALIDATE',
+    function: 'XUESSO3',
```
Comment (creindle) [inline emulation/models/xus-esso-validate.js]: The (devdocs)[http://vistadataproject.info/artifacts/devdocs/VISTARPC/XUS_ESSO_VALIDATE] for this didn't have a specific method listed. What should I write for the function?

File: emulation/models/xus-iam-bind-user.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const RPC_EMULATOR_MODEL = [{
+    id: 'XUS_IAM_BIND_USER',
+    name: 'XUS IAM BIND USER',
+    function: 'XUESSO3',
```
Comment (creindle) [inline emulation/models/xus-iam-bind-user.js]: The [devdocs](http://vistadataproject.info/artifacts/devdocs/VISTARPC/XUS_IAM_BIND_USER) didn't have a specific method listed. What should I write here?

File: emulation/models/xus-av-code.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    complexity: 'HIGH',
```
Comment (creindle) [inline emulation/models/xus-av-code.js]: I had to set a complexity for these models, so I set them to 'HIGH' for now. They seem like difficult RPCs.

File: emulation/models/xus-esso-validate.js (hunk 2 above)
Comment (mfuroyama) [reply emulation/models/xus-esso-validate.js]: It's not a big dea, but for now, let's just switch it to `ESSO^XUESSO3`

File: emulation/models/xus-iam-bind-user.js (hunk 3 above)
Comment (mfuroyama) [reply emulation/models/xus-iam-bind-user.js]: `IAMBU^XUESSO3` even though the routine isn't in the docs

File: emulation/models/xus-av-code.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    returnType: 'ARRAY',
+    inputParameters: [{
+        id: 'AVCODE',
+        parameterType: 'LITERAL',
+        description: 'accessCode_";"_verifyCode in unencrypted form.',
+        required: true,
+        example: '',
+    }],
+    invokeRPC(rpcArgs, session, callback) {
+    },
```
Comment (mfuroyama) [inline emulation/models/xus-av-code.js]: Sorry, I should have been more clear about this, but could we do one final thing here?  Could we have all the `invokeRPC` functions call the `callback` function with a generic enumeratred error saying something like "Emulation not implemented" or something?

We *might* also want to consider putting that 'unimplemented stub' function into the model utils, maybe call it `notImplemented`, so in the models, we can have:

`
...
   invokeRPC: notImplemented,
...
```

File: emulation/models/modelUtils.js [hunk 6]
```diff
@@ -1,6 +1,9 @@
 'use strict';
 
 const _ = require('lodash');
+const BaseService = require('../../services/baseService');
```
Comment (mfuroyama) [inline emulation/models/modelUtils.js]: Remove this please

File: emulation/models/modelUtils.js [hunk 7]
```diff
@@ -134,6 +137,13 @@ const isNilOrFalse = val => (_.isNil(val) || (_.isBoolean(val) && !val));
  */
 const createLocationEntity = (locationId, session) => (+locationId ? createURNFromParams(44, locationId, session.stationId) : 'LOC');
 
+/**
+ * Creates stub RPC models
+ * @param  {Function}  callback  Function called on process completion, with standard (err, result) signature.
+ * @param  {Function}  callback  Function called on process completion, with generic enumerated error 'Emulation not implemented'
+ */
+const notImplemented = cb => cb(null, new Error('Not implemented'));
```
Comment (mfuroyama) [inline emulation/models/modelUtils.js]: Let's make this function signature as follows:
```
const notImplemented = (args, session, cb) => cb(new Error('Not implemented'));
```

This way, in the RPC model, we can just do this:
```
   invokeRPC: notImplemented,
```

File: emulation/models/modelUtils.js [hunk 8]
```diff
@@ -134,6 +134,13 @@ const isNilOrFalse = val => (_.isNil(val) || (_.isBoolean(val) && !val));
  */
 const createLocationEntity = (locationId, session) => (+locationId ? createURNFromParams(44, locationId, session.stationId) : 'LOC');
 
+/**
+ * Creates stub RPC models
+ * @param  {Function}  callback  Function called on process completion, with standard (err, result) signature.
+ * @param  {Function}  callback  Function called on process completion, with generic enumerated error 'Emulation not implemented'
```
Comment (mfuroyama) [inline emulation/models/modelUtils.js]: LOL whut?

Comment (creindle) [reply emulation/models/modelUtils.js]: Brain fart. I'm sorry.

**Pull request merged**